arbor/arbexcept.cpp

```diff
@@ -16,6 +16,31 @@ bad_cell_description::bad_cell_description(cell_kind kind, cell_gid_type gid):
     kind(kind)
 {}
 
+bad_target_description::bad_target_description(cell_gid_type gid, cell_size_type rec_val, cell_size_type cell_val):
```

Comment:
These error messages will be reported to users who might not be writing a recipe (e.g. using a tool that generates the recipe for them), so I think it is good to present the error information as generally as possible, e.g:
`Model building error on cell 23: connection target 108 is out of range. Cell 23 has 88 targets, in the range [0:87].`

Reply:
Okay, I'll write more useful messages.